pythonforandroid/recipes/python3/patches/fix-ctypes-util-find-library.patch

```diff
@@ -10,14 +10,31 @@ diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
 +if True:
 +    def find_library(name):
 +        # Check the user app libs and system libraries directory:
-+        app_root = os.path.normpath(os.path.abspath('../../'))
-+        lib_search_dirs = [os.path.join(app_root, 'lib'), "/system/lib"]
-+        for lib_dir in lib_search_dirs:
-+            for filename in os.listdir(lib_dir):
-+                if filename.endswith('.so') and (
-+                        filename.startswith("lib" + name + ".") or
-+                        filename.startswith(name + ".")):
-+                    return os.path.join(lib_dir, filename)
++        import re
++        from jnius import autoclass
++        activity = autoclass("org.kivy.android.PythonActivity").mActivity
++        package_name = activity.getApplicationContext().getPackageName()
++        manager = activity.getApplicationContext().getPackageManager()
++        manager_class = autoclass("android.content.pm.PackageManager")
++        native_lib_dir = manager.getApplicationInfo(
++            package_name, manager_class.GET_SHARED_LIBRARY_FILES
++        ).nativeLibraryDir
++        lib_search_dirs = [native_lib_dir, "/system/lib"]
++        for lib_dir in [l for l in lib_search_dirs if os.path.exists(l)]:
++            # Filter file names so given name="mymodule.so" we match one of:
```

Comment:
Thanks for adding this comment.
I actually thought `find_library()` was more used with `name="mylib"` rather than `name="mylib.so"`

Reply:
oops yeah I guess that part of the comment is wrong :joy:

Reply:
fixed it now :joy:

Reply:
Thanks!